Src/ReadBarometer.c

```diff
@@ -6,127 +19,226 @@
+        // Read the third byte (bits 7-0)
+        HAL_SPI_TransmitReceive(&hspi2, &READ_BYTE_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
         pressureReading += dataInBuffer;
 
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_SET);
 
-        // Read D2 temperature
+        /* Read Digital Temperature (D2) -------------------------------------*/
+
+        // Tell the barometer to convert the temperature to a digital value with an over-sampling ratio of 512
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_RESET);
         HAL_SPI_Transmit(&hspi2, &ADC_D2_512_CONV_CMD, CMD_SIZE, CMD_TIMEOUT);
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_SET);
 
-        osDelay(2); // 1.17ms max conversion time for OSR 512
+        osDelay(2); // 1.17ms max conversion time for an over-sampling ratio of 512
 
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_RESET);
-        HAL_SPI_Transmit(&hspi2, &ADC_READ_CMD, CMD_SIZE, CMD_TIMEOUT);
 
-        temperatureReading = 0;
+        HAL_SPI_Transmit(&hspi2, &ADC_READ_CMD, CMD_SIZE, CMD_TIMEOUT);
 
+        // Read the first byte (bits 23-16)
         HAL_SPI_TransmitReceive(&hspi2, &ADC_READ_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
-        temperatureReading += dataInBuffer << 16;
+        temperatureReading = dataInBuffer << 16;
 
+        // Read the second byte (bits 15-8)
         HAL_SPI_TransmitReceive(&hspi2, &ADC_READ_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
         temperatureReading += dataInBuffer << 8;
 
+        // Read the third byte (bits 7-0)
         HAL_SPI_TransmitReceive(&hspi2, &ADC_READ_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
         temperatureReading += dataInBuffer;
 
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_SET);
 
-        // calculate 1st order pressure and temperature (1st order algorithm from MS5607-02BA03 datasheet)
-        double dT = temperatureReading - c5Tref * pow(2, 8);
-        double temp = 2000 + dT * c6Tempsens / pow(2, 23);
-        double off = c2Off * pow(2, 17) + dT * c4Tco / pow(2, 6);
-        double sens = c1Sens * pow(2, 16) + dT * c3Tcs / pow(2, 7);
-        // pressure and temperature were not divided by 100 to keep the decimal places
-        double pressure = (pressureReading * sens / pow(2, 21) - off) / pow(2, 15); // need to divide P by 100 to get mbar
-        double temperature = (temp); // need to divide T by 100 to get degrees Celcius
+        /* Calculate First-Order Temperature and Pressure --------------------*/
+
+        int32_t dT      = temperatureReading - (c5Tref << 8);
+        int32_t temp    = 2000 + ((dT * c6Tempsens) >> 23); // Divide this value by 100 to get degrees Celcius
+        int64_t off     = (c2Off << 17) + ((dT * c4Tco) >> 6);
+        int64_t sens    = (c1Sens << 16) + ((dT * c3Tcs) >> 7);
+        // int32_t p    = (((pressureReading * sens) >> 21) - off) >> 15;
```

Comment:
I think we can take these 2 comments out. If we wanted to get the first order pressure, it would make more sense to just comment out the calculation for the second order calculations and calculate the pressure below it.

Reply:
Done